### PR TITLE
Geometry tests 1

### PR DESCRIPTION
**Pull request by Athight**
Checking in windows setup and a few tests to check style and whatnot to reduce the risk of wasted future effort.

Let me know:
- if there are any preferred styles
- whether this is sufficient test coverage in general
- whether this style of pull request makes sense
- how large would the ideal pull request be?

Cheers :)

### CODE REVIEW

File: tests/test_geometry_common.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        closest_geometry = common.closest(geom, targets, 1)
+        self.assertEqual(closest_geometry, LineString([(-0.5, 0), (0.5, 0)]))
+
+    def test_closest_collection_two_mixed_targets(self):
+        geom = GeometryCollection([Point(0, 0), LineString([(1, 0), (1, 1)])])
+        targets = [Point(0.5, 0.5), LineString([(0, 2.1), (2, 0)])]
+        closest_geometry = common.closest(geom, targets, 1)
+        self.assertEqual(closest_geometry, LineString([(0, 2.1), (2, 0)]))
+
+    # This just confirms an edge case, if this changes, this test can be updated
```
Comment (allfed-dev) [inline tests/test_geometry_common.py]: Good pickup - it'd be nice if there was consistent behaviour for cases like this, but it seems like this would be hard to implement. Happy to have this as a slightly quirky expected edge case for now.

**Comment by allfed-dev** [comment]
Looks good! re: your questions:

> if there are any preferred styles

What you've got is fine

> whether this is sufficient test coverage in general

Yes, more than sufficient. It's often a judgement call how much certain behaviour should be covered by libs like `shapely` rather than explicitly tested here, and you seem to have got the balance right. 

> whether this style of pull request makes sense

Yep this is great, really good work. In future however, make a branch on this repo instead of forking the repo.

> how large would the ideal pull request be?

Keeping it to less than a few hundred lines of code is ideal, I think a PR for testing a single related set of functions is good (e.g. testing `common.py`), but you may want to split it up more than that where there's lots of behaviour to test. Trust your judgement on this one.
